apis/Google.Cloud.Metadata.V1/Google.Cloud.Metadata.V1.IntegrationTests/Emulator/metadata_server.py

```diff
@@ -18,13 +18,135 @@
 from werkzeug.wrappers import Request
 import common, errno, getopt, json, logging, os, socket, string, sys, time
 
-numeric_project_id = 12345
-numeric_project_id_token = u'<NUMERIC_PROJECT_ID>'
-project_id = u'fake-project'
-project_id_token = u'<PROJECT_ID>'
+class ServerState:
+  fake_numeric_project_id = 12345
+  fake_project_id = u'fake-project'
+
+  _credential_scopes = ["https://www.googleapis.com/auth/cloud-platform",
+                        "https://www.googleapis.com/auth/userinfo.email"]
+
+  def __init__(self, is_for_testing, project_id, numeric_project_id, use_default_credentials, credentials_files):
+    if project_id:
+      self.project_id = project_id
+    elif is_for_testing:
+      self.project_id = ServerState.fake_project_id
+    else:
+      self.project_id = os.environ.get('TEST_PROJECT', ServerState.fake_project_id)
+
+    if numeric_project_id:
+      self.numeric_project_id = numeric_project_id
+    else:
+      self.numeric_project_id = ServerState.fake_numeric_project_id
```

Comment:
I do not like baking these sorts of details into the class itself...ServerState should be about, well, server state. It shouldn't really have the details about a fake project.

I think the better pattern is to have a constant server state that you initialize with these values...so instead of relying on them to be set if there is no numeric_project_id, you have one that is set to this.

That sort of depends on how it is used, but I think in this case that makes more sense. It's ideal not to entangle that if possible.